test/Dialect/FIRRTL/lowerXMR.mlir

```diff
@@ -47,6 +47,26 @@ firrtl.circuit "Top" {
 
 // -----
 
+// Test 0-width xmrs are handled
+// CHECK-LABEL: firrtl.circuit "Top" {
+firrtl.circuit "Top" {
+  firrtl.module @Top(in %bar_a : !firrtl.ref<uint<0>>, in %bar_b : !firrtl.ref<vector<uint<0>,10>>) {
+    %a = firrtl.wire : !firrtl.uint<0>
+    %0 = firrtl.ref.resolve %bar_a : !firrtl.ref<uint<0>>
+    // CHECK:  %c0_ui0 = firrtl.constant 0 : !firrtl.uint<0>
```

Comment:
Nit: doing a capture instead of the hard-coded `%c0_ui0` would be better. (I don't expect we would ever change this. It would be a nightmare on the testing side if we did!)